File: frontend/src/modules/graph/application/use-cases/stageFile.ts
import { WorkingChangesWriter } from "@modules/graph/application/contracts/WorkingChangesWriter";

export async function stageFile(
  workingChangesWriter: WorkingChangesWriter,
  repositoryPath: string,
  filePath: string
): Promise<void> {
  return workingChangesWriter.stageFile(repositoryPath, filePath);
}

File: backend/src/app/working_changes/stage_file.rs
use std::path::Path;

use crate::app::working_changes::contracts::WorkingChangesWriter;
use crate::domain::errors::AppError;

pub fn stage_file(
    working_changes_writer: &dyn WorkingChangesWriter,
    repository_path: &Path,
    file_path: &str,
) -> Result<(), AppError> {
    working_changes_writer.stage_file(repository_path, file_path)
}
